packages/dataclass-editor/src/components/dataclass/detail/DataClassDetailContent.tsx

```diff
@@ -14,31 +14,22 @@ import {
 } from '@axonivy/ui-components';
 import { useAppContext } from '../../../context/AppContext';
 import type { DataClass } from '../data/dataclass';
-import { classTypeOf } from '../data/dataclass-utils';
+import {
+  classTypeOf,
+  handleClassTypeChange,
+  handleDataClassEntityPropertyChange,
+  handleDataClassPropertyChange,
+  isEntity
+} from '../data/dataclass-utils';
 import { AnnotationsTable } from './AnnotationsTable';
 
 export const DataClassDetailContent = () => {
   const { dataClass, setDataClass } = useAppContext();
 
   const classType = classTypeOf(dataClass);
 
-  const handleClassTypeChange = (classType: string) => {
-    const newDataClass = structuredClone(dataClass);
-    newDataClass.isBusinessCaseData = false;
-    newDataClass.entity = undefined;
-    if (classType === 'BUSINESS_DATA') {
-      newDataClass.isBusinessCaseData = true;
-    } else if (classType === 'ENTITY') {
-      newDataClass.entity = { tableName: '' };
-    }
-    setDataClass(newDataClass);
-  };
-
-  const handleDataClassPropertyChange = <DKey extends keyof DataClass>(key: DKey, value: DataClass[DKey]) => {
-    const newDataClass = structuredClone(dataClass);
-    newDataClass[key] = value;
-    setDataClass(newDataClass);
-  };
+  const onPropertyChange = <DKey extends keyof DataClass>(key: DKey, value: DataClass[DKey]) =>
+    handleDataClassPropertyChange(key, value, dataClass, setDataClass);
```

Comment:
why you don't write custom hooks where you can access the dataClass and setDataClass directly instead of always add them as properties...